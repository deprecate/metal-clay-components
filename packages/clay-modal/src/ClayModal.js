import 'clay-button';
import 'clay-icon';
import {Config} from 'metal-state';
import {EventHandler} from 'metal-events';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import dom from 'metal-dom';
import Soy from 'metal-soy';

import templates from './ClayModal.soy.js';

const KEY_CODE_ESC = 27;

/**
 * Metal ClayModal component.
 */
class ClayModal extends Component {
	/**
	 * @inheritDoc
	 */
	created() {
		this.overlayElement_ = this.valueOverlayElementFn_();
		this.eventHandler_ = new EventHandler();
	}

	/**
	 * @inheritDoc
	 */
	attached() {
		this.addListener('click', this.handleDocumentClick_.bind(this), true);
		this.addListener('hide', this.defaultHideModal_, true);
		this.addListener('show', this.defaultShowModal_, true);
		this.addListener(
			'touchend',
			this.handleDocumentClick_.bind(this),
			true
		);
		this.addListener('transitionend', this.handleTransitionEnd_, true);
	}

	/**
	 * @inheritDoc
	 */
	detached() {
		super.detached();
		this.eventHandler_.removeAllListeners();
	}

	/**
	 * @inheritDoc
	 */
	syncVisible() {
		if (this.visible) {
			dom.enterDocument(this.overlayElement_);
			this.overlayElement_.offsetHeight;
			dom.addClasses(this.overlayElement_, 'show');
		} else {
			dom.exitDocument(this.overlayElement_);
		}
	}

	/**
	 * @inheritDoc
	 */
	syncIsTransitioning_() {
		if (this.isTransitioning_ && !this.visible) {
			this.isTransitioning_ = false;
			this.visible = true;
		} else if (this.isTransitioning_ && this.visible) {
			this.visible = false;
		}
	}

	/**
	 * Close modal and remove transitions.
	 * @private
	 */
	defaultHideModal_() {
		dom.removeClasses(this.overlayElement_, 'show');
		dom.removeClasses(document.body, 'modal-open');

		this.isTransitioning_ = true;

		this.eventHandler_.removeAllListeners();
	}

	/**
	 * Open modal and add transition.
	 * @private
	 */
	defaultShowModal_() {
		dom.addClasses(document.body, 'modal-open');

		this.isTransitioning_ = true;

		this.eventHandler_.add(
			dom.on(document, 'keyup', this.handleKeyup_.bind(this))
		);
	}

	/**
	 * Handle the click buttons and emit event.
	 * @param {!Event} event
	 * @private
	 */
	handleClickFooterButton_(event) {
		this.emit('clickButton', event);
	}

	/**
	 * Handle the footer close button click and emits events.
	 * @param {!Event} event
	 * @private
	 */
	handleClickCloseButtonFooter_(event) {
		this.emit('clickButton', event);
		this.emit('hide');
	}

	/**
	 * Handle the click and emit close modal.
	 * @private
	 */
	handleCloseModal_() {
		this.emit('hide');
	}

	/**
	 * Check if contain click target event in element and emit close modal.
	 * @param {!Event} event
	 * @private
	 */
	handleDocumentClick_(event) {
		if (dom.contains(event.target, this.element)) {
			this.emit('hide');
		}
	}

	/**
	 * Handle click key code esc and emit close modal.
	 * @param {!Event} event
	 * @private
	 */
	handleKeyup_(event) {
		if (event.keyCode === KEY_CODE_ESC) {
			this.emit('hide');
		}
	}

	/**
	 * Handle css transition ends.
	 * @param {!Event} event
	 * @private
	 */
	handleTransitionEnd_(event) {
		if (
			event.target === this.element &&
			this.isTransitioning_ &&
			!this.visible
		) {
			this.isTransitioning_ = false;
		}
	}

	/**
	 * Create fragment of the overlay modal.
	 * @return {Element} The resulting document fragment.
	 * @private
	 */
	valueOverlayElementFn_() {
		return dom.buildFragment('<div class="modal-backdrop fade"></div>')
			.firstChild;
	}

	/**
	 * Emit open modal.
	 * @public
	 */
	show() {
		this.emit('show');
	}
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayModal.STATE = {
	/**
	 * Body of the modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|html|undefined}
	 * @default undefined
	 */
	body: Config.any(),

	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * Buttons of the footer.
	 * @instance
	 * @memberof ClayModal
	 * @type {?array|undefined}
	 * @default undefined
	 */
	footerButtons: Config.arrayOf(
		Config.shapeOf({
			alignment: Config.oneOf(['left', 'right']).value('right'),
			label: Config.string().required(),
			style: Config.oneOf([
				'borderless',
				'link',
				'primary',
				'secondary',
				'unstyled',
			]),
			type: Config.oneOf(['button', 'close', 'reset', 'submit']).value(
				'button'
			),
		})
	),

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 * Adds transitions when show is called.
	 * @instance
	 * @memberof ClayModal
	 * @type {?bool}
	 * @default false
	 * @private
	 */
	isTransitioning_: Config.bool()
		.value(false)
		.internal(),

	/**
	 * The size of element modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	size: Config.oneOf(['full-screen', 'lg', 'sm']),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	spritemap: Config.string(),

	/**
	 * Status messages.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	status: Config.oneOf(['danger', 'info', 'success', 'warning']),

	/**
	 * Title of the modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	title: Config.string(),

	/**
	 * Url to place an iframe in the body of the modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	url: Config.string(),

	/**
	 * Modal visible when show is called.
	 * @instance
	 * @memberof ClayModal
	 * @type {?bool}
	 * @default false
	 * @private
	 */
	visible: Config.bool().value(false),
};

defineWebComponent('clay-modal', ClayModal);

Soy.register(ClayModal, templates);

export {ClayModal};
export default ClayModal;
