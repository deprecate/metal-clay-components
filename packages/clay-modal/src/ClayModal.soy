{namespace ClayModal}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param? _handleClickButtonFooter: any}
	{@param? _handleCloseModal: any}
	{@param? _isTransitioning: bool}
	{@param? body: string|html}
	{@param? footer: list<?>}
	{@param? id: string}
	{@param? iframe: bool}
	{@param? size: string}
	{@param? spritemap: string}
	{@param? status: string}
	{@param? title: string}
	{@param? visible: bool}

	{let $classesSize kind="text"}
		modal-dialog
		{if $size}
			{sp}modal-{$size}
		{elseif $iframe}
			{sp}modal-full-screen
		{/if}

		{if $status}
			{sp}modal-{$status}
		{/if}
	{/let}

	{let $attributes kind="attributes"}
		class="
			fade modal
			{if $_isTransitioning}
				{sp}show
			{/if}
		"

		{if $visible}
			style="display: block;"
		{/if}

		{if $id}
			id="{$id}"
		{/if}

		role="dialog"

		tabindex="-1"
	{/let}

	<div {$attributes}>
		<div class="{$classesSize}">
			<div class="modal-content">
				{call .header}
					{param _handleCloseModal: $_handleCloseModal /}
					{param spritemap: $spritemap /}
					{param title: $title /}
				{/call}

				{call .body}
					{param body: $body /}
					{param iframe: $iframe /}
				{/call}

				{call .footer}
					{param _handleClickButtonFooter: $_handleClickButtonFooter /}
					{param _handleCloseModal: $_handleCloseModal /}
					{param footer: $footer /}
				{/call}
			</div>
		</div>
	</div>
{/template}

/**
 * Render a header of the modal.
 */
{template .header}
	{@param? _handleCloseModal: any}
	{@param? spritemap: string}
	{@param? title: string}

	{if $title and $spritemap}
		<div class="modal-header">
			<h4 class="modal-title">{$title}</h4>

			{call ClayButton.render}
				{param elementClasses: 'close' /}
				{param events: ['click': $_handleCloseModal] /}
				{param icon: [
					'alignment': 'left',
					'monospaced': false,
					'spritemap': $spritemap,
					'symbol': 'times'
				] /}
				{param style: 'secondary' /}
			{/call}
		</div>
	{/if}
{/template}

/**
 * Render a body of the modal.
 */
{template .body}
	{@param? body: string|html}
	{@param? iframe: bool}

	{let $classes kind="text"}
		modal-body
		{if $iframe}
			{sp}modal-body-iframe
		{/if}
	{/let}

	<div class="{$classes}">
		{$body}
	</div>
{/template}

/**
 * Render a footer of the modal.
 */
{template .footer}
	{@param? _handleClickButtonFooter: any}
	{@param? _handleCloseModal: any}
	{@param? footer: list<?>}

	{if $footer}
		<div class="modal-footer">
			<div class="modal-item-first">
				{foreach $item in $footer}
					{if not $item.type}
						{call ClayButton.render}
							{param events: ['click': $_handleClickButtonFooter] /}
							{param label: $item.label /}
							{param style: $item.style ?: 'primary' /}
							{param type: 'button' /}
						{/call}
						{sp}
					{/if}
				{/foreach}
			</div>
			<div class="modal-item"></div>
			<div class="modal-item-last">
				{foreach $item in $footer}
					{if $item.type == 'close'}
						{call ClayButton.render}
							{param elementClasses: 'close-modal' /}
							{param events: ['click': $_handleCloseModal] /}
							{param label: $item.label /}
							{param style: 'secondary' /}
							{param type: 'button' /}
						{/call}
						{sp}
					{elseif $item.type == 'save'}
						{call ClayButton.render}
							{param events: ['click': $_handleClickButtonFooter] /}
							{param label: $item.label /}
							{param style: 'primary' /}
							{param type: 'button' /}
						{/call}
						{sp}
					{/if}
				{/foreach}
			</div>
		</div>
	{/if}
{/template}
