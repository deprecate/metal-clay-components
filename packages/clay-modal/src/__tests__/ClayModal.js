import ClayModal from '../ClayModal';

let clayModal;
let body = '<h4>Hello world!</h4>';
let spritemap = 'icons.svg';

describe('ClayModal', function() {
  afterEach(() => {
    if (clayModal) {
      clayModal.dispose();
    }
  });

  it('should generate the default markup', () => {
    clayModal = new ClayModal();

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with title', () => {
    clayModal = new ClayModal({
      title: 'My title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with title and header closing button', () => {
    clayModal = new ClayModal({
      spritemap: spritemap,
      title: 'My title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with body', () => {
    clayModal = new ClayModal({
      body: 'My body',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with footer buttons', () => {
    clayModal = new ClayModal({
      footerButtons: [
        {
          label: 'Cancel',
          type: 'close',
        },
        {
          label: 'Save',
        },
      ],
    });

    expect(clayModal).toMatchSnapshot();
  });

  // eslint-disable-next-line
  it('should render a modal with footer buttons aligned to left and right', () => {
    clayModal = new ClayModal({
      footerButtons: [
        {
          label: 'Cancel',
          type: 'close',
        },
        {
          label: 'Save',
        },
        {
          alignment: 'left',
          label: 'My button 1',
        },
        {
          alignment: 'left',
          label: 'My button 2',
        },
      ],
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with iframe as body', () => {
    clayModal = new ClayModal({
      url: 'http://www.claycss.com',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with status `danger` and title', () => {
    clayModal = new ClayModal({
      status: 'danger',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with status `info` and title', () => {
    clayModal = new ClayModal({
      status: 'info',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with status `success` and title', () => {
    clayModal = new ClayModal({
      status: 'success',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with status `warning` and title', () => {
    clayModal = new ClayModal({
      status: 'warning',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with size `small`', () => {
    clayModal = new ClayModal({
      body: body,
      size: 'sm',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with size `large`', () => {
    clayModal = new ClayModal({
      body: body,
      size: 'lg',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal with size `full-screen`', () => {
    clayModal = new ClayModal({
      body: body,
      size: 'full-screen',
      spritemap: spritemap,
      title: 'My Title',
    });

    expect(clayModal).toMatchSnapshot();
  });

  it('should render a modal opened', () => {
    clayModal = new ClayModal({
      body: body,
      spritemap: spritemap,
      title: 'My Title',
      visible: true,
    });

    expect(clayModal).toMatchSnapshot();
  });
});
