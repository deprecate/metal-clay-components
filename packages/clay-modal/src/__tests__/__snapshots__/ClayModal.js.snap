// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayModal should generate the default markup 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-body"></div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a full-screen modal 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-full-screen">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a large modal 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-lg">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal complete 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal open 1`] = `
<div class=" fade modal show" role="dialog" tabindex="-1" style="display: block;">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with body and footer 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with buttons in position right and left in footer 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first">
          <button class="btn btn-primary" type="button">All</button>
          <button class="btn btn-secondary" type="button">All</button>
        </div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with header and body 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with iframe 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-full-screen">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body modal-body-iframe">
        <iframe title="Clay" src=""></iframe>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with status \`danger\` 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-danger">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with status \`info\` 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-info">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with status \`success\` 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-success">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a modal with status \`warning\` 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-warning">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayModal should render a small modal 1`] = `
<div class=" fade modal" role="dialog" tabindex="-1">
  <div class="modal-dialog modal-sm">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">My title</h4>
        <button class="btn close btn-secondary" aria-label="times">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
            <use xlink:href="icons.svg#times"></use>
          </svg>
        </button>
      </div>
      <div class="modal-body">
        <h4>Hello world!</h4>
      </div>
      <div class="modal-footer">
        <div class="modal-item-first"></div>
        <div class="modal-item"></div>
        <div class="modal-item-last">
          <button class="btn close-modal btn-secondary" type="button">Cancel</button>
          <button class="btn btn-primary" type="button">Save</button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
